src/as/math/Color.ts

```diff
@@ -227,24 +225,29 @@ export class Color{
 
 	readonly isColor: boolean = true
 
-	// TODO see https://github.com/AssemblyScript/assemblyscript/issues/645
-	// set(color: Color): Color
-	// set(color: f32): Color
-	// set(color: string): Color
-	// set<T>(color: T): this {
-	// 	if (color instanceof Color) {
-	// 		this.copy(color)
-	// 	} else if (isInteger<u32>(color)) {
-	// 		this.setHex(color)
-	// 	}
-	// 	// TODO, setStyle uses RegExp, and AS doesn't have RegExp yet.
-	// 	// else if (isString<T>(color)) {
-	// 	// 	this.setStyle(color)
-	// 	// }
-	// 	// else throw new Error('Color.set: invalid arg')
+	/**
+	 * Allows multiple ways of creating a color. F.e. `new
+	 * Color().set(otherColor)`, `new Color(r, g, b)`, `new
+	 * Color().set('blue')`, `new Color().set('#ff6600')`, or `new
+	 * Color().set(0xff6600)`
+	 */
+	// set(color: Color): this
+	// set(color: i32): this
+	// set(r: f32, g: f32, b: f32): this
+	// set(color: string): this TODO
+	set<T>(colorOrR: T, g: f32 = 0, b: f32 = 0): this {
+		if (colorOrR instanceof Color) this.copy(colorOrR)
+		else if (isInteger(colorOrR)) this.setHex(colorOrR)
+		else if (isFloat(colorOrR)) this.setRGB(f32(colorOrR), g, b)
+		// TODO, no strings yet, because setStyle uses RegExp, and AS doesn't
```

Comment:
regex pattern matching can be bit slow, we could match with bits .. possibly. just an idea

Reply:
Sure. We can always optimize. Will be sweet to have perf tests with as-bench.